packages/turbo-vsc/ARCHITECTURE.md

```diff
@@ -0,0 +1,153 @@
+# Architecture
+
+This document attempts to give a high level overview to the extension / LSP,
+which APIs it uses to achieve its feature set, and how it is structured.
+
+## Client
+
+You're here! The client is the side that runs in VSCode. It is essentially
+an entry point into the LSP but there are a few other things it manages
+mostly for convience sake.
+
+- basic syntax highlighting for the pipeline gradient
+- discovery and installation of global / local turbo
+- toolbar item to enable / disable the daemon
+- some editor commands
+  - start deamon
+  - stop daemon
+  - restart daemon
+  - run turbo command
+  - run turbo lint
+
+Otherwise it simply selects the correct LSP binary and runs it using vscode's
+built-in LSP library, and the LSP in turn interacts with the turbo daemon to
+get the information it needs to fulfil client requests.
+
+## Daemon
+
+The daemon plays a minor role in relaying important metadata about the
```

Comment:
Can you link "The daemon" to the daemon crate (if we have one of those)?

Reply:
Daemon just lives in turborepo-lib atm